### PR TITLE
Introduce bindings for ExternalWorkload resources

### PR DESCRIPTION
**Pull request by mateiidavid**
We introduced an ExternalWorkload CRD for mesh expansion. This change follows up by adding bindings for Rust and Go code.

For Go code:
* We add a new schema and ExternalWorkload types
* We also update the code-gen script to generate informers
* We add a new informer type to our abstractions built on-top of client-go, including a function to check if a client has access to the resource.

For Rust code:
* We add ExternalWorkload bindings to the policy controller.

### CODE REVIEW

File: policy-controller/k8s/api/src/external_workload.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct Condition {
+    /// Type of the condition
+    // see: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle#pod-conditions
+    #[serde(rename = "type")]
+    typ: String,
+    /// Status of the condition.
+    /// Can be True, False, Unknown
+    status: String,
```
Comment (alpeb) [inline policy-controller/k8s/api/src/external_workload.rs:78]: Nit: can we have an enum for these possible values?

**Review by alpeb — APPROVED** [review]
This looks good to me :+1:

File: controller/gen/apis/externalworkload/v1alpha1/types.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+}
+
+// MeshTls describes TLS settings associated with an external workload
+type MeshTls struct {
+	// Identity associated with the workload. Used by peers to perform
+	// verification in the mTLS handshake
+	Identity string `json:"identity,omitempty"`
+	// ServerName is the DNS formatted name associated with the workload. Used
+	// to terminate TLS using the SNI extension.
+	ServerName string `json:"serverName,omitempty"`
```
Comment (zaharidichev) [inline controller/gen/apis/externalworkload/v1alpha1/types.go:66]: We need to do another pass of thinking what is optional and what is not here, but for now I think this is good.

**Review by zaharidichev — APPROVED** [review]
LGTM

**Pull request merged**